### ISSUE TITLE
How to deal with included tags?

### ISSUE THREAD

**Issue by hightechguy** [marko-js/htmljs-parser#76]
Hi,

I'm trying to parse
```
<ol>
	<li><h1>my title</h1></li>
	<li>my string</li>
</ol>
```
and get li tags innert HTML i.e:
```
[
    "<h1>my title</h1>",
    "my string"
]
```
What would be the the good approach to get included tags inner HTML?

Thanks!

**Comment by DylanPiercey** [comment]
@hightechguy for Marko we use this module to build an object structure (AST) and then we do analysis on that.

What are you trying to use htmljs-parser for?

**Comment by hightechguy** [comment]
I'm adopting web page editor for the client needs, something like one we use to write these messages.
Currently <ul>/<ol> are in work. There is no problem to validate and get <li> data when innerHTML of <li>  is simple text, but users may add other HTML tags in, as h1 in my example. I would like to parse <ul>,  get each <li> innerHTML and pass it to a text field for editing

**Comment by hightechguy** [comment]
Hi, Dylan

Can you please address my original question?

Thanks!

**Comment by DylanPiercey** [comment]
@hightechguy this is not really the intent of the HTMLJSParser. However, within the events you always have access to the `parser` object which contains `pos` (the current parser position).

Combine that with the `endPos` for a tag and you can slice the string to get it's innerHTML.

eg:

```
var openPositionStack = [];

...
onOpenTag(ev) {
  if (!ev.selfClosed) {
   openPositionStack(ev.endPos);
  }
}
onCloseTag(ev, parser) {
  if (!ev.selfClosed) {
    var startPos = openPositionStack.pop();
    var innerHTML = code.slice(startPos, ev.pos);
  }
}
```

Or something similar to the above anyway.

**Comment by mlrawlings** [comment]
> this is not really the intent of the HTMLJSParser

@hightechguy Just to clarify what Dylan means, this parser is designed to parse the HTML-like [syntax of Marko](https://markojs.com/docs/syntax/).  If you're looking to parse just plain HTML, you might also want to check out [`htmlparser2`](https://github.com/fb55/htmlparser2) which has a similar event based api to this module, but parses standard HTML.  We actually used to use `htmlparser2` in Marko v2 before we started introducing custom language constructs.
